maryana_khoptyak/Task_2/retry_decorator.py

```diff
@@ -0,0 +1,37 @@
+"""
+The module implements 'retry' decorator.
+If some part of code in decorated function fails, it will be re-run again for number of retries that is passed
+as decorator parameter or until function executes successfully.
+If decorated function executes successfully, it will not be re-run.
+"""
+
+import random
+list_of_fruits = ("banana", "strawberry", "lemon", "pineapple", "apple", "coconut", "watermelon")
+FAVORITE_FRUIT = "strawberry"
+
+def decorator_with_params(retries):
+    def decorator(func):
+        def wrapper(list_with_items):
+            choice = func(list_with_items)
+            if choice != FAVORITE_FRUIT:
+                for _ in range(retries):
+                    choice = func(list_with_items)
+                    if choice == FAVORITE_FRUIT:
```

Comment:
it looks too complicated with odd logic here.

The main idea of such retry decorator is to handle some exception and re-run some functionality N times.

You're not handling any error here, just doing something strange with your fruits.